app/src/main/java/com/google/android/soundchecker/AudioEncoderDecoderActivity.kt

```diff
@@ -546,8 +546,8 @@ class AudioEncoderDecoderActivity : ComponentActivity() {
                     ",bitsPerSample: " + reader.getBitsPerSample() +
                     ",numSampleFrames: " + reader.getNumSampleFrames())
 
-            mOutputTrackCount++
-            val currentOutputTrackCount = mOutputTrackCount
+            mPlayAudioMonotonicCounter++
```

Comment:
Is this running in a different thread then onStop()?
If so then if the onStop() ran before line 550 then there may be a race condition and the loop will not exit.
Safer to have only ONE thread write to a variable. So may need two variables, each written by their own thread.
For example a mStartRequestCounter and a mStopRequestCounter.

    while (startRequestCounter > stopRequestCounter) {}

Reply:
Since this is a rare race condition for a test app, I'm inclined to leave this as is. Thanks!

Reply:
Ack